controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java

```diff
@@ -739,6 +744,25 @@ public List<DataplaneTokenVersions> readDataplaneTokens(TenantName tenantName) {
         return readSlime(dataplaneTokenPath(tenantName)).map(DataplaneTokenSerializer::fromSlime).orElse(List.of());
     }
 
+    // -------------- Endpoint certificate pool -------------------------------
+
+    public void addToCertificatePool(String s, EndpointCertificateMetadata metadata, String pool) {
+        curator.set(certificatePoolRoot.append(pool).append(s), asJson(EndpointCertificateMetadataSerializer.toSlime(metadata)));
+    }
+
+    public void removeFromCertificatePool(String s, String pool) {
+        curator.delete(certificatePoolRoot.append(pool).append(s));
+    }
+
+    public Map<String, EndpointCertificateMetadata> readCertificatePool(String pool) {
```

Comment:
I think it would be better to create a concrete pair-type to represent a certificate and it's state, e.g. `record PooledCertificate(EndpointCertificateMetadata, PooledCertificate.State)` and have this method return `List<PooledCertificate>`.

(I also think `EndpointCertificateMetadata` could be simplified to `EndpointCertificate`, but we can do that later)